File: DemoApp-Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/data/main/ListItem.java
package com.applovin.enterprise.apps.demoapp.data.main;

/**
 * All the list items on main screen should confirm to this interface.
 * <p>
 * Created by santoshbagadi on 2019-09-10.
 */
public interface ListItem
{
    int TYPE_SECTION_HEADER = 0;
    int TYPE_AD_ITEM        = 1;

    /**
     * @return The type of list item.
     */
    int getType();
}

File: DemoApp-Kotlin/app/src/main/java/com/applovin/enterprise/apps/demoapp/data/home/ListItem.kt
package com.applovin.enterprise.apps.demoapp.data.home

/**
 * Interface for all list items on the home screen.
 * <p>
 * Created by Harry Arakkal on 9/17/2019
 */
interface ListItem
{
    companion object
    {
        const val SECTION_HEADER = 0
        const val AD_ITEM = 1
    }

    val type: Int
}
